File: rust/src/models/connection_type.rs
// this file is @generated
use std::fmt;

use serde::{Deserialize, Serialize};

/// Type of connection between platform and connected account
#[derive(
    Clone, Copy, Debug, Default, PartialEq, Eq, PartialOrd, Ord, Hash, Serialize, Deserialize,
)]
pub enum ConnectionType {
    #[default]
    #[serde(rename = "standard")]
    Standard,

    #[serde(rename = "express")]
    Express,
}

impl fmt::Display for ConnectionType {
    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
        let value = match self {
            Self::Standard => "standard",
            Self::Express => "express",
        };
        f.write_str(value)
    }
}

File: typescript/src/models/connectionType.ts
// this file is @generated
/** Type of connection between platform and connected account */
export enum ConnectionType {
  Standard = "standard",
  Express = "express",
}

export const ConnectionTypeSerializer = {
  _fromJsonObject(object: any): ConnectionType {
    return object;
  },

  _toJsonObject(self: ConnectionType): any {
    return self;
  },
};
